
Socket permission changes to remove world permissions form the system bus socket



By default, and in a way that cannot be overridden by configuration, the dbus-daemon
attempts to reset the socket permissions to 0777.  This patch amends that to 0660.

diff -Nbru dbus-1.8.16/dbus/dbus-sysdeps-unix.c dbus-1.8.16-new/dbus/dbus-sysdeps-unix.c
--- dbus-1.8.16/dbus/dbus-sysdeps-unix.c	2015-02-03 15:47:02.000000000 +0000
+++ dbus-1.8.16-new/dbus/dbus-sysdeps-unix.c	2015-11-11 18:14:01.553632424 +0000
@@ -1138,8 +1138,8 @@
   /* Try opening up the permissions, but if we can't, just go ahead
    * and continue, maybe it will be good enough.
    */
-  if (!abstract && chmod (path, 0777) < 0)
-    _dbus_warn ("Could not set mode 0777 on socket %s\n",
+  if (!abstract && chmod (path, 0660) < 0)
+    _dbus_warn ("Could not set mode 0660 on socket %s\n",
                 path);
 
   return listen_fd;
